ui/dashboards/src/context/DashboardProvider/panel-group-delete-slice.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { getPanelKeyFromRef } from '@perses-dev/core';
+import { StateCreator } from 'zustand';
+import { Middleware } from './common';
+import { PanelGroupId, PanelGroupSlice } from './panel-group-slice';
+import { PanelSlice } from './panel-slice';
+
+/**
+ * Slice that handles the visual editor state and related actions for deleting a Panel Group.
+ */
+export interface PanelGroupDeleteSlice {
+  deletePanelGroupDialog?: DeletePanelGroupDialog;
+
+  /**
+   * Delete panel group and all the panels within the group
+   */
+  deletePanelGroup: (panelGroupId: PanelGroupId) => void;
+
+  openDeletePanelGroupDialog: (panelGroupId: PanelGroupId) => void;
+  closeDeletePanelGroupDialog: () => void;
+}
+
+export interface DeletePanelGroupDialog {
```

Comment:
Could rename to `DeletePanelGroupDialogState` like `DeletePanelDialogState` but it's mouthful for sure